pkg/util/test/rw2.go

```diff
@@ -99,32 +99,38 @@ func AddHistogramSeries(
 type SymbolTableBuilder struct {
 	count   uint32
 	symbols map[string]uint32
+	offset  uint32
 }
 
 func NewSymbolTableBuilder(symbols []string) *SymbolTableBuilder {
+	return NewSymbolTableBuilderWithOffset(symbols, 0)
+}
+
+func NewSymbolTableBuilderWithOffset(symbols []string, offset uint32) *SymbolTableBuilder {
 	symbolsMap := make(map[string]uint32)
 	for i, sym := range symbols {
-		symbolsMap[sym] = uint32(i)
+		symbolsMap[sym] = uint32(i) + offset
 	}
 	return &SymbolTableBuilder{
 		count:   uint32(len(symbols)),
 		symbols: symbolsMap,
+		offset:  offset,
 	}
 }
 
 func (symbols *SymbolTableBuilder) GetSymbol(sym string) uint32 {
 	if i, ok := symbols.symbols[sym]; ok {
 		return i
 	}
-	symbols.symbols[sym] = symbols.count
+	symbols.symbols[sym] = symbols.offset + symbols.count
 	symbols.count++
-	return symbols.count - 1
+	return symbols.offset + symbols.count - 1
 }
 
 func (symbols *SymbolTableBuilder) GetSymbols() []string {
 	res := make([]string, len(symbols.symbols))
 	for sym, i := range symbols.symbols {
```

Comment:
This logic looks broken to me. The map iteration order is random in golang. Each time you may return symbols in a different order.

Reply:
That's okay; the map is (string->index). We use the map values to reconstruct the order, we don't care about the iteration. It may look like {"b": 1, "a": 0, "c": 2} and come back in that order, and we'd still put "a" in position 0.

All this modulo a shift by `offset`. It's probably a naming thing that's misleading; `i` looks like the index of the iteration but it's not.

Reply:
Sorry, I was drunk when I read the code :)